### ISSUE TITLE
Generation using StableDiffusionPipeline is not deterministic

### ISSUE THREAD

**Issue by nreHieW** [huggingface/diffusers#5931]
### Describe the bug

Generating using the same seed and generator outputs different latents. Tested on `stabilityai/stable-diffusion-2-base` and `stabilityai/stable-diffusion-2-1`

Used CPU generator from [docs](https://huggingface.co/docs/diffusers/using-diffusers/reproducibility)

### Reproduction

```python
import torch
from diffusers import StableDiffusionPipeline, DPMSolverMultistepScheduler

model_id = "stabilityai/stable-diffusion-2-base"

pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe = pipe.to("cuda")

height = pipe.unet.config.sample_size * pipe.vae_scale_factor
width = pipe.unet.config.sample_size * pipe.vae_scale_factor
latents = pipe.prepare_latents(1, pipe.unet.config.in_channels, height, width, pipe.unet.dtype, pipe.device, generator = torch.Generator(device = "cpu").manual_seed(42))

prompt = "a photo of an astronaut riding a horse on mars"
imgs_1 = pipe(prompt, output_type="latent", num_images_per_prompt = 1, generator = torch.Generator(device = "cpu").manual_seed(42), latents = latents.clone()).images


imgs_2 = pipe(prompt, output_type="latent", num_images_per_prompt = 1, generator = torch.Generator(device = "cpu").manual_seed(42), latents = latents.clone()).images

print(torch.allclose(imgs_1, imgs_2)) # False
```

### System Info

- `diffusers` version: 0.20.2
- Platform: Linux-5.15.0-46-generic-x86_64-with-glibc2.31
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.33.0
- Accelerate version: 0.22.0
- xFormers version: not installed
- Using GPU in script?: Yes A4000
- Using distributed or parallel set-up in script?: No

### Who can help?

@yiyixuxu @DN6

**Comment by shauray8** [comment]
@nreHieW I could not reproduce any anomalies regarding the output. maybe try upgrading the version, given you are using ```0.20.2```Try upgrading it to the latest version

**Comment by nreHieW** [comment]
@shauray8 thanks! that fixed it not sure why 👍

**Comment by miunshine** [comment]
Hi,
I am getting a similar issue when using 'stabilityai/stable-diffusion-xl-base-1.0'
My current diffusers version is '0.24.0.dev0' 
Torch version '2.0.1'

I create a generator and use it:
```
generator = torch.manual_seed(33)

base(prompt=prompt, num_inference_steps=10, guidance_scale=7.0, generator=generator).images[0].save("out.png")
```

The output image is different each time, unless I redefine the generator again after each generation.

```
generator = torch.manual_seed(33)
base(prompt=prompt, num_inference_steps=10, guidance_scale=7.0, generator=generator).images[0].save("out.png")

generator = torch.manual_seed(33)
base(prompt=prompt, num_inference_steps=10, guidance_scale=7.0, generator=generator).images[0].save("out.png")
```
This produces the same image.

**Comment by DN6** [comment]
Hi @nreHieW check out my response here related to determinism with GPUs
https://github.com/huggingface/diffusers/issues/5523
